timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/watch/ArrayLockEventSlidingWindow.java

```diff
@@ -39,10 +39,16 @@ long lastVersion() {
         return nextSequence - 1;
     }
 
-    void add(LockWatchEvent.Builder eventBuilder) {
+    /**
+     * Returns the {@link LockWatchEvent} that was replaced if the buffer is already full.
+     */
+    Optional<LockWatchEvent> add(LockWatchEvent.Builder eventBuilder) {
```

Comment:
I'm a little hesitant to keep this named as `add`, maybe `addOrReplace` or something similar?

Reply:
I moved the metrics to `ArrayLockEventSlidingWindow`, so this, once again, does not return anything. Should we still rename it?

Reply:
nah